tachyon/src/main/java/org/apache/zeppelin/tachyon/TachyonInterpreter.java

```diff
@@ -0,0 +1,181 @@
+                .build());
+  }
+
+  @Override
+  public void open() {
+    logger.info("Starting Tachyon shell to connect to " + tachyonMasterHostname +
+      " on port " + tachyonMasterPort);
+
+    System.setProperty(TACHYON_MASTER_HOSTNAME, tachyonMasterHostname);
+    System.setProperty(TACHYON_MASTER_PORT, tachyonMasterPort);
+    tfs = new TfsShell(new TachyonConf());
+  }
+
+  @Override
+  public void close() {
+    logger.info("Closing Tachyon shell");
+
+    try {
+      tfs.close();
+    } catch (IOException e) {
+      logger.error("Cannot close connection", e);
+    }
+  }
+
+  @Override
+  public InterpreterResult interpret(String st, InterpreterContext context) {
+    String[] lines = splitAndRemoveEmpty(st, "\n");
+    return interpret(lines, context);
+  }
+  
+  private InterpreterResult interpret(String[] commands, InterpreterContext context) {
+    boolean isSuccess = true;
+    totalCommands = commands.length;
+    completedCommands = 0;
+    
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    PrintStream ps = new PrintStream(baos);
+    PrintStream old = System.out;
+    
+    System.setOut(ps);
+    
+    for (String command : commands) {
+      String[] args = splitAndRemoveEmpty(command, " ");
+      int commandResuld = tfs.run(args);
+      System.out.println();
+      if (commandResuld != 0) {
+        isSuccess = false;
+        break;
+      } else {
+        completedCommands += 1;
+      }
+    }
+
+    System.out.flush();
+    System.setOut(old);
+    
+    if (isSuccess) {
+      return new InterpreterResult(Code.SUCCESS, baos.toString());
+    } else {
+      return new InterpreterResult(Code.ERROR, baos.toString());
+    }
+  }
+  
+  private String[] splitAndRemoveEmpty(String st, String splitSeparator) {
+    String[] voices = st.split(splitSeparator);
+    ArrayList<String> result = new ArrayList<String>();
+    for (String voice : voices) {
+      if (!voice.trim().isEmpty()) {
+        result.add(voice);
+      }
+    }
+    return result.toArray(new String[result.size()]);
+  }
+
+  private String[] splitAndRemoveEmpty(String[] sts, String splitSeparator) {
+    ArrayList<String> result = new ArrayList<String>();
+    for (String st : sts) {
+      result.addAll(Arrays.asList(splitAndRemoveEmpty(st, splitSeparator)));
+    }
+    return result.toArray(new String[result.size()]);
+  }
+
+  @Override
+  public void cancel(InterpreterContext context) { }
+
+  @Override
+  public FormType getFormType() {
+    return FormType.NATIVE;
+  }
+
+  @Override
+  public int getProgress(InterpreterContext context) {
+    return completedCommands * 100 / totalCommands;
+  }
+
+  @Override
+  public List<String> completion(String buf, int cursor) {
+    String[] words = splitAndRemoveEmpty(splitAndRemoveEmpty(buf, "\n"), " ");
+    String lastWord = words[ words.length - 1 ];
+    ArrayList<String> voices = new ArrayList<String>();
+    for (String command : keywords) {
```

Comment:
Instead of loop through keywords list, could keywords be in form of HashSet so you could just check HashSet.contains? 
Does it need to check with startsWith?